sdk/storage/azure-storage-datamovement/inc/azure/storage/datamovement/datamovement_options.hpp

```diff
@@ -3,22 +3,64 @@
 
 #pragma once
 
+#include <functional>
+#include <memory>
 #include <string>
 
+#include <azure/core/credentials/credentials.hpp>
+#include <azure/storage/common/storage_credential.hpp>
+
 namespace Azure { namespace Storage {
   struct StorageTransferManagerOptions final
   {
     std::string TransferStateDirectoryPath;
+    Nullable<int> NumThreads;
+    Nullable<size_t> MaxMemorySize;
   };
 
-  namespace Blobs {
+  struct TransferCredential final
+  {
+    std::string ConnectionString;
```

Comment:
Minor: Is connection string still necessary?  The other three should be able to represent all credentials we support for now.

Reply:
Just like customers can create clients from connection string, I think they can do it here. I'm considering making factory methods here, so that they can only provide one type of authorization.